core/src/main/java/org/frankframework/pipes/IfPipe.java

```diff
@@ -0,0 +1,407 @@
+@Forward(name = "else", description = "the configured condition is not met")
+@EnterpriseIntegrationPattern(EnterpriseIntegrationPattern.Type.ROUTER)
+public class IfPipe extends AbstractPipe {
+
+	private String elseForwardName = "else";
+	private String thenForwardName = "then";
+	private PipeForward elseForward;
+	private PipeForward thenForward;
+
+	private String namespaceDefs = null;
+	private boolean namespaceAware = XmlUtils.isNamespaceAwareByDefault();
+	private TransformerPool transformerPool;
+	private String jsonPathExpression = null;
+	private String xpathExpression = null;
+	private String expressionValue = null;
+
+	private String regex = null;
+	private int xsltVersion = XmlUtils.DEFAULT_XSLT_VERSION;
+
+	private DefaultMediaType defaultMediaType = DefaultMediaType.XML;
+
+	public enum DefaultMediaType {
+		XML(MediaType.APPLICATION_XML),
+		JSON(MediaType.APPLICATION_JSON);
+
+		final MediaType mediaType;
+
+		DefaultMediaType(MediaType mediaType) {
+			this.mediaType = mediaType;
+		}
+	}
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		thenForward = assertExistsAndGetForward(thenForwardName);
+		elseForward = assertExistsAndGetForward(elseForwardName);
+
+		if (StringUtils.isNotEmpty(xpathExpression)) {
+			transformerPool = TransformerPool.configureTransformer0(this, namespaceDefs, xpathExpression, null,
+					TransformerPool.OutputType.XML, false, getParameterList(), xsltVersion
+			);
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		MimeType mimeType = getMimeType(message);
+
+		// Determine the media type and if the correct *pathExpression was provided
+		if (jsonPathExpression != null && !MediaType.APPLICATION_JSON.equals(mimeType)
+				|| xpathExpression != null && !MediaType.APPLICATION_XML.equals(mimeType)) {
+			throw new PipeRunException(this, "Incorrect pathExpression provided for given mediaType + " + mimeType);
+		}
+
+		PipeForward pipeForward = determineForward(message, session);
+
+		log.debug("resolved forward [{}] to path [{}]", pipeForward.getName(), pipeForward.getPath());
+
+		return new PipeRunResult(pipeForward, message);
+	}
+
+	private MimeType getMimeType(Message message) {
+		MimeType computedType = MessageUtils.computeMimeType(message);
+
+		if (computedType != null) {
+			// check if computedType is one of JSON or XML, if not, fall through
+			boolean matchesOneOfSupportedTypes = Arrays.stream(DefaultMediaType.values())
+					.anyMatch(supportedType -> supportedType.mediaType.equals(computedType));
+
+			if (matchesOneOfSupportedTypes) {
+				return computedType;
+			}
+		}
+
+		// Default to
+		return defaultMediaType.mediaType;
+	}
+
+	/**
+	 * Based on the pipe settings, tries to evaluate the expression and determine the correct forward
+	 */
+	private PipeForward determineForward(Message message, PipeLineSession session) throws PipeRunException {
```

Comment:
We might be able to extend here easily for Yaml in the future. Perhaps even extract the code to specific implementation classes for the diffferent media types. I'll create a follow up ticket for this